ext/digest/digest.h

```diff
@@ -64,10 +64,4 @@ rb_id_metadata(void)
     return rb_intern_const("metadata");
 }
 
-static inline VALUE
-rb_digest_make_metadata(const rb_digest_metadata_t *meta)
-{
-#undef RUBY_UNTYPED_DATA_WARNING
-#define RUBY_UNTYPED_DATA_WARNING 0
-    return rb_obj_freeze(Data_Wrap_Struct(0, 0, 0, (void *)meta));
-}
+VALUE rb_digest_make_metadata(const rb_digest_metadata_t *meta);
```

Comment:
This doesn't work on all platforms.
`rb_ext_resolve_symbol` is added for the case like this.
Also, older version of digest didn't provide this function, this is a binary incompatibility.

Reply:
`rb_digest_make_metadata` is `static inline` now and it will be inlined in each extension library.
But this PR moves it to digest.c.
The point is that a symbol in an extension library may not be referred from another extension library.